code/ship/ship.cpp

```diff
@@ -3291,10 +3302,12 @@ int parse_ship_values(ship_info* sip, const bool is_template, const bool first_t
 		Error(LOCATION, "%s '%s' has too many primary banks (%d).  Maximum for ships is currently %d.\n", info_type_name, sip->name, sip->num_primary_banks, MAX_SHIP_PRIMARY_BANKS);
 	}
 
+	memset(sip->allowed_weapons, 0, sizeof(int) * MAX_WEAPON_TYPES);
```

Comment:
Yes, although no amount of defensive coding can protect you against accidentally iterating through a for-loop too many times, and it's not memsetting too much memory *now*, is it?

If just this one line gets changed, and then somebody actually does change the size of the `allowed_weapons` array, I think having all-but-one reference breaking to be less useful than just having all of them break. I'd rather have the consistency than introduce better practices piecemeal, especially since it wouldn't be that much effort to just fix all of them at once.